test.js

```diff
@@ -126,6 +126,7 @@ test('supports Armenian', t => {
 });
 
 test('leading underscore', t => {
-	t.is(slugify('_foo bar', {leadingUnderscore: true}), '_foo-bar');
-	t.is(slugify('_foo_bar', {leadingUnderscore: true}), '_foo-bar');
+	t.is(slugify('_foo bar', {preserveLeadingUnderscore: true}), '_foo-bar');
+	t.is(slugify('_foo_bar', {preserveLeadingUnderscore: true}), '_foo-bar');
+	t.is(slugify('__foo__bar', {preserveLeadingUnderscore: true}), '_-foo-bar');
```

Comment:
This should result in:

```suggestion
	t.is(slugify('__foo__bar', {preserveLeadingUnderscore: true}), '_foo-bar');
```

Would also be good to add a test for `slugify('____-___foo__bar', {preserveLeadingUnderscore: true})`, which should also result in `_foo-bar`.

Reply:
I was testin and after the 
```js
string = string.replace(patternSlug, separator);
```
in all these examples it became _\\-foo\\-bar.

So, I thought in just slicing the separator (\\-) after it. Something like:

```js
if(string.slice(1,2) == separator){
		string = string.slice(0, 1) + string.slice(2, string.length);
	}
``` 

it works, but what do you think?

Reply:
I think you're necessarily complicating the implementation. You could just do:

```js
const shouldPrependUnderscore = options.preserveLeadingUnderscore && string.startsWith('_');
```

at the start and then prepend a underscore at the end if `shouldPrependUnderscore` is true.

Reply:
I actually thought that way firstly, but i thought would be better to just modify the current regex and got confused afterall

Reply:
just committed the changes, sindre.